src/Test.Utilities.Sarif/HttpMockHelper.cs

```diff
@@ -116,6 +116,13 @@ private static bool CompareHeaders(HttpRequestHeaders headers1, HttpRequestHeade
 
         protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
```

Comment:
>SendAsync [](http://example.com/codeflow?start=53&length=9)

There is an easier change. Update this class to contain an array/list of response. Maintain a counter of calls to SendAsync and simply return the response associated with that index.

In other words, return 'mockedResponse[callCount++]'

That's it. Now you can delete the other items from the response tuple. Now we can stop matching as below.

The reason this works is that in *all* our tests, the calls to SendAsync are sequential and ordered.

Reply:
This approach came to me first but I was not sure 
1. the possibility of Async call will return same order as executed, e.g. the first land on a core that is more congested than a second one.
2. tests to be executed in parallel

@michaelcfanning  let me know

Reply:
Every mock helper should be created per rule that is under test. I think all our rules effectively make these calls synchronously on the same thread, i.e., they await the result of the first before moving on. So i think the approach i suggest will be ok.

Reply:
1. I have fixed the other 2 comments about using CreateXXX method. CodeFlow freeze if I try to reply them.
2. I have tried using 3 ways: maintain a counter with List, Queue without counter, ConcurrentQueue without counter,
only ConcurrentQueue works better than the other two(many errors in the other repo), 
because it is async and async methods typically span on a different thread to avoid blocking the calling thread. 
3. **please check with the PR that actually applies it, in the other repo**.